go/protocols/flow/flow.proto

```diff
@@ -419,6 +419,85 @@ message MaterializationSpec {
   protocol.JournalSpec recovery_log_template = 6;
 }
 
+// OAuth2Spec describes an OAuth2 provider
+message OAuth2Spec {
+  // Name of the OAuth2 provider
+  string provider = 1;
```

Comment:
nit: Is this just a human-readable display name, or is it used as a machine-readable key at some point in the process? It would be nice to clarify in the comment.

Reply:
This is a machine-readable key, used in the UI to distinguish which button to show. Will clarify